AddNodeScene vm updates to support all async fetch request

gemdev111 opened the pull request:
#105

gemdev111 committed:
AddNodeScene vm updates to support all async fetch request

gemcoder21 commented on Gem/Nodes/ViewModels/AddNodeSceneViewModel.swift:
```diff
@@ -89,13 +93,16 @@ extension AddNodeSceneViewModel {
         do {
-            let (latency, chainId) = try await fetchChainID(service: service)
-            let nodeInfo = try await fetchAdditionalNodeInfo(service: service)
+            async let (requestLatency, netowkId) = fetchChainID(service: service)
```
```suggestion
            async let (requestLatency, networkId) = fetchChainID(service: service)
```

gemdev111 committed:
Typo fixed netowkId => networkId

gemdev111 committed:
Minor Refactored AddNodeViewModel
Adjusted AddNodeScene & vm, added AddNodeResultViewModel to omit optionals unwrapping
